logp/core_test.go

```diff
@@ -212,3 +216,117 @@ func TestLoggingECSFields(t *testing.T) {
+	// temporary directory cannot be removed if a file inside it is still
+	// open.
+	//
+	// To circumvent this problem on Windows we use os.MkdirTemp
+	// leaving it behind and delegating to the OS the responsibility
+	// of cleaning it up (usually on restart).
+	if runtime.GOOS == "windows" {
+		var err error
+		tempDir1, err = os.MkdirTemp("", t.Name()+"-*")
+		if err != nil {
+			t.Fatalf("could not create temporary directory: %s", err)
+		}
+		tempDir2, err = os.MkdirTemp("", t.Name()+"-*")
+		if err != nil {
+			t.Fatalf("could not create temporary directory: %s", err)
+		}
+	} else {
+		// We have no problems on Linux and Darwin, so we can rely on t.TempDir
+		// that will remove the files once the tests finishes.
+		tempDir1 = t.TempDir()
+		tempDir2 = t.TempDir()
+	}
+
+	expectedLogMessage := "this is a log message"
+	expectedLogLogger := t.Name() + "-second"
+
+	// We follow the same approach as on a Beat, first the logger
+	// (always global) is configured and used, then we instantiate
+	// a new one, secondLogger, and perform the tests on it.
+	loggerCfg := DefaultConfig(DefaultEnvironment)
+	loggerCfg.Beat = t.Name() + "-first"
+	loggerCfg.ToFiles = true
+	loggerCfg.ToStderr = false
+	loggerCfg.Files.Name = "test-log-file-first"
+	// We want a separate directory for this logger
+	// and we don't need to inspect it.
+	loggerCfg.Files.Path = tempDir1
+
+	// Configures the global logger with the "default" log configuration.
+	if err := Configure(loggerCfg); err != nil {
+		t.Errorf("could not initialise logger: %s", err)
+	}
+	logger := L()
+
+	// Create a log entry just to "test" the logger
+	logger.Info("not the message we want")
+	if err := logger.Sync(); err != nil {
+		t.Fatalf("could not sync log file from fist logger: %s", err)
+	}
+
+	// Actually clones the logger and use the "WithFileOutput" function
+	secondCfg := DefaultConfig(DefaultEnvironment)
+	secondCfg.ToFiles = true
+	secondCfg.ToStderr = false
+	secondCfg.Files.Name = "test-log-file"
+	secondCfg.Files.Path = tempDir2
+
+	// We do not call Configure here as we do not want to affect
+	// the global logger configuration
+	secondLogger := NewLogger(t.Name() + "-second")
+	secondLogger = secondLogger.WithOptions(zap.WrapCore(WithFileOrStderrOutput(secondCfg)))
+	secondLogger.Info(expectedLogMessage)
+	if err := secondLogger.Sync(); err != nil {
+		t.Fatalf("could not sync log file from second logger: %s", err)
+	}
+
+	// Writes again with the first logger to ensure it has not been affected
+	// by the new configuration on the second logger.
+	logger.Info("not the message we want")
+	if err := logger.Sync(); err != nil {
+		t.Fatalf("could not sync log file from fist logger: %s", err)
+	}
+
+	// Find the log file. The file name gets the date added, so we list the
+	// directory and ensure there is only one file there.
+	files, err := os.ReadDir(tempDir2)
+	if err != nil {
+		t.Fatalf("could not read temporary directory '%s': %s", tempDir2, err)
+	}
```

Comment:
Shouldn't you also check that the first logger still works? This test would still pass if cloning the first logger broke it, or if writes to the second logger also went into the files of the first logger.

Reply:
Creating a new logger is not a new behaviour, hence I didn't think about looking at it's log  file/output. But I did ensure that if the first logger is used after the second is created and has its output replaced, the first one won't log to the same file as the second one.

Do you still believe we need to add another case? Like also reading/validating the log file for the first logger?

Reply:
I added assertions to the original/default logger as well: b672556